src/fmu/tools/qcforward/_blockedwells_vs_gridprops.py

```diff
@@ -0,0 +1,328 @@
+                print(comb[comb["WELLNAME"].isin(wells)].to_string())
+            elif "warn" in self.ldata.show_data:
+                dfr_warn_or_stop = dfr[dfr["STATUS"] != "OK"]
+                wells = list(dfr_warn_or_stop["WELL"])
+                if "all" in wells:
+                    wells.remove("all")
+                if wells:
+                    print(f"Wells with WARN or STOP are these {wells}:")
+                    print(comb[comb["WELLNAME"].isin(wells)].to_string())
+                else:
+                    print("No wells reported WARN or STOP")
+
+            else:
+                print(f"Wells with WARN or STOP or OK are:")
+                print(comb.to_string())
+
+        if status == "STOP":
+            QCC.force_stop("STOP criteria is found!")
+
+    def compare_bw_props(self) -> pd.DataFrame:
+        """Given data, do a comparison of blcked wells cells vs props, via XTGeo."""
+
+        # dataframe for the blocked wells
+        dfbw = self.gdata.bwells.get_dataframe()
+        if self._gdata.project is not None:
+            # when parsing blocked wells from RMS, cell indices starts from 0, not  1
+            dfbw["I_INDEX"] += 1
+            dfbw["J_INDEX"] += 1
+            dfbw["K_INDEX"] += 1
+
+        # dataframe for the properties, need some processing (column names)
+        dfprops = self.gdata.gridprops.get_dataframe(ijk=True, grid=self.gdata.grid)
+        dfprops = dfprops.rename(
+            columns={"IX": "I_INDEX", "JY": "J_INDEX", "KZ": "K_INDEX"}
+        )
+
+        # merge the dataframe on I J K index
+        comb = pd.merge(
+            dfbw,
+            dfprops,
+            how="inner",
+            on=["I_INDEX", "J_INDEX", "K_INDEX"],
+            suffixes=("__bw", "__model"),  # in case the names are equal -> add suffix
+        )
+        QCC.print_debug("Made a combined dataframe!")
+        QCC.print_debug(f"\n {comb}")
+
+        diffs = {}
+
+        # compare the relevant properties
+        for bwprop, modelprop in self._ldata.compare.items():
+            usebwprop = bwprop
+            usemodelprop = modelprop
+            if bwprop == modelprop:
+                usebwprop = bwprop + "__bw"
+                usemodelprop = modelprop + "__model"
+            dname = bwprop + ":" + modelprop
+            dnameflag = dname + "_flag"
+            comb = self._eval_tolerance(comb, usebwprop, usemodelprop, dname, dnameflag)
+            diffs[dname] = dnameflag
+
+        return self._evaluate_diffs(comb, diffs), comb
+
+    def _eval_tolerance(self, df_in, bwprop, modelprop, diffname, diffnameflag):
+        """Make a flag log for diffs based on tolerance input."""
+        comb = df_in.copy()
+        tol = self.ldata.tolerance
+
+        relative = False
+        if isinstance(tol, float):
+            tolerance = tol
+        elif isinstance(tol, dict):
+            if "rel" in tol.keys():
+                relative = True
+            tolerance = list(tol.values())[0]
```

Comment:
Might be better to have relative as an oneliner
`relative = isinstance(tol, dict) and "rel" in tol`

This could also be done for the tolerance, but less readable maybe 🙂 
`tolerance = tol if isinstance(tol, float) else list(tol.values)[0]`

Reply:
Yes some regards ternary if (one-liners) as less readable